core/src/main/java/com/google/errorprone/bugpatterns/ThrowSpecificExceptions.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2019 The Error Prone Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.errorprone.bugpatterns;
+
+import static com.google.errorprone.BugPattern.ProvidesFix.REQUIRES_HUMAN_ATTENTION;
+import static com.google.errorprone.BugPattern.SeverityLevel.WARNING;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker.NewClassTreeMatcher;
+import com.google.errorprone.fixes.SuggestedFix;
+import com.google.errorprone.fixes.SuggestedFixes;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.Matchers;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.NewClassTree;
+
+/** Bugpattern to discourage throwing base exception classes.. */
+@BugPattern(
+    name = "ThrowSpecificExceptions",
+    summary =
+        "Consider throwing more specific exceptions rather than (e.g.) RuntimeException. Throwing"
+            + " generic exceptions forces any users of the API that wish to handle the failure"
+            + " mode to catch very non-specific exceptions that convey little information.",
+    providesFix = REQUIRES_HUMAN_ATTENTION,
+    severity = WARNING)
+public final class ThrowSpecificExceptions extends BugChecker implements NewClassTreeMatcher {
+  private static final ImmutableList<AbstractLikeException> ABSTRACT_LIKE_EXCEPTIONS =
+      ImmutableList.of(
+          AbstractLikeException.of(RuntimeException.class, IllegalStateException.class),
+          AbstractLikeException.of(Throwable.class, AssertionError.class),
+          AbstractLikeException.of(Error.class, AssertionError.class));
```

Comment:
Just curious: was `Exception.class` left out on purpose? It may not have an obvious replacement, but Error Prone could still flag its use. WDYT?